sharing/resource_share.go

```diff
@@ -176,6 +186,11 @@ func (beforeSi ShareInfo) Diff(afterSi ShareInfo) []ShareDataChange {
 func ResourceShare() common.Resource {
 	shareSchema := common.StructToSchema(ShareInfo{}, func(m map[string]*schema.Schema) map[string]*schema.Schema {
 		m["name"].DiffSuppressFunc = common.EqualFoldDiffSuppress
+		m["object"].Set = func(i any) int {
+			objectStruct := i.(map[string]any)
+			hashString := objectStruct["name"].(string) + objectStruct["data_object_type"].(string)
```

Comment:
Should we add any separator between these fields?

@nkvuong do you remember - can we share the same object with different `shared_as`? Or if we can share same table with different `history_data_sharing_status`?